# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/jsapi/TransceiverImpl.cpp
# Commit: 3ca24e761cc3
# Full Hash: 3ca24e761cc31bbb71c659661874a995ef3d3db1
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 6
# Description:
#   Bug 1654112 - Remove WebrtcGmpPCHandleSetter and plumb pc handles explicitly. r=bwc
#   
#   With the libwebrtc update codecs will be created off-main.
#   WebrtcGmpPCHandleSetter not only adds a lot of indirection, it is also
#   main-thread-only. With this patch we can create gmp codecs on any thread.
# ==============================================================================

diff -r 1913c332695e -r 3ca24e761cc3 dom/media/webrtc/jsapi/TransceiverImpl.cpp
--- a/dom/media/webrtc/jsapi/TransceiverImpl.cpp	Wed Feb 10 09:58:11 2021 +0100
+++ b/dom/media/webrtc/jsapi/TransceiverImpl.cpp	Wed Feb 10 10:34:14 2021 +0100
@@ -73,8 +73,6 @@
     return;
   }
 
-  mConduit->SetPCHandle(mPCHandle);
-
   mReceiver = new RTCRtpReceiver(aWindow, aPrivacyNeeded, aPCHandle,
                                  aTransportHandler, aJsepTransceiver,
                                  aMainThread, aStsThread, mConduit, this);
@@ -145,7 +143,7 @@
 }
 
 void TransceiverImpl::InitVideo() {
-  mConduit = VideoSessionConduit::Create(mCallWrapper, mStsThread);
+  mConduit = VideoSessionConduit::Create(mCallWrapper, mStsThread, mPCHandle);
 
   if (!mConduit) {
     MOZ_MTLOG(ML_ERROR, mPCHandle << "[" << mMid << "]: " << __FUNCTION__
@@ -465,7 +463,6 @@
 
   if (mJsepTransceiver->mSendTrack.SetJsConstraints(constraints)) {
     if (mTransmitPipeline->Transmitting()) {
-      WebrtcGmpPCHandleSetter setter(mPCHandle);
       DebugOnly<nsresult> rv = UpdateConduit();
       MOZ_ASSERT(NS_SUCCEEDED(rv));
     }